crates/compiler/test_derive/src/decoding.rs

```diff
@@ -162,15 +162,23 @@ fn record_2_fields() {
                                   rest: #Derived.rec2.rest
                                 })
                       _ -> Skip
-                  \#Derived.stateRecord ->
-                    when #Derived.stateRecord.first is
+                  \#Derived.stateRecord, #Derived.fmt ->
+                    when when #Derived.stateRecord.first is
+                        Ok #Derived.first -> Ok #Derived.first
+                        _ ->
+                          when decodeWith [] decoder #Derived.fmt is
+                            #Derived.decRec2 -> #Derived.decRec2.result is
                       Ok #Derived.first ->
-                        when #Derived.stateRecord.second is
+                        when when #Derived.stateRecord.second is
+                            Ok #Derived.second -> Ok #Derived.second
+                            _ ->
+                              when decodeWith [] decoder #Derived.fmt is
+                                #Derived.decRec -> #Derived.decRec.result is
```

Comment:
Let's be very explicit about matching this only if the result is `Err NotFound`. If the field fails to decode, that's not the same as it being missing and attempting an optional decode.

Reply:
As I understand it failing to decode a field results in the decoding stopping and the finalizer never being run. But I've made the change anyway for clarity and future proofing.